presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/DigestAndPercentileStateSerializer.java

```diff
@@ -41,27 +41,29 @@ public void serialize(DigestAndPercentileState state, BlockBuilder out)
             out.appendNull();
         }
         else {
-            DynamicSliceOutput sliceOutput = new DynamicSliceOutput(state.getDigest().estimatedSerializedSizeInBytes() + SIZE_OF_DOUBLE);
-            // write digest
-            state.getDigest().serialize(sliceOutput);
-            // write percentile
-            sliceOutput.appendDouble(state.getPercentile());
+            Slice serialized = state.getDigest().serialize();
 
-            Slice slice = sliceOutput.slice();
-            VARCHAR.writeSlice(out, slice);
+            DynamicSliceOutput output = new DynamicSliceOutput(serialized.length() + SIZE_OF_DOUBLE);
```

Comment:
you are missing the "INT"... also you don't need a dynamic output when you know the length.  `Slices.allocate(length).getOutput()`.  At the end verify `SliceOutput.getSize` is the size you wanted